wl/server/Server.java

```diff
@@ -0,0 +1,253 @@
+		connectionProps.put("password", Utils.config("databasepassword"));
+		java.sql.Connection database =
+		    java.sql.DriverManager.getConnection​(Utils.config("databasename"), connectionProps);
+
+		System.out.println("[" + new Date() + "] Server starting...");
+		ServerSocket serverSocket = new ServerSocket(Integer.valueOf(Utils.config("port")));
+		new Thread() {
+			public void run() {
+				boolean errored = false;
+				do try {
+						Calendar nextSync = Calendar.getInstance();
+						nextSync.set(Calendar.HOUR_OF_DAY, 3);
+						nextSync.set(Calendar.MINUTE, 0);
+						nextSync.set(Calendar.SECOND, 0);
+						nextSync.set(Calendar.MILLISECOND, 0);
+						long now = System.currentTimeMillis();
+						long then = nextSync.getTimeInMillis();
+						while (then < now + 60000) then += 1000 * 60 * 60 * 24;
+						System.out.println("Next GitHub sync scheduled for " + new Date(then) +
+						                   " (" + Utils.durationString(then - now) +
+						                   " remaining).");
+						Thread.sleep(then - now);
+						System.out.println("Waking up for GitHub sync at " + new Date());
+						synchronized (syncer) {
+							System.out.println("Cleaning up inactive threads at " + new Date());
+							syncer.check();
+							if (errored)
+								throw new Exception(
+								    "You still have not resolved the merge conflicts. "
+								    + "Please do so soon!");
+							System.out.println("Performing GitHub sync at " + new Date());
+							Utils._staticprofiles.clear();
+							syncer.sync();
+						}
+					} catch (Exception e) {
+						errored = true;
+						System.out.println("GitHub sync ERROR: " + e);
+						try {
+							String str;
+							String msg = "@Noordfrees\n\n"
+							             + "The automated GitHub sync on the server "
+							             + "has failed with the following error message:\n"
+							             + "```\n" + e + "\n```\n\n```\n$ git status";
+							Process p = Runtime.getRuntime().exec(
+							    new String[] {"bash", "-c", "git status"});
+							p.waitFor();
+							BufferedReader b =
+							    new BufferedReader(new InputStreamReader(p.getInputStream()));
+							while ((str = b.readLine()) != null) msg += "\n" + str;
+							msg += "\n```\n\nThe automated syncs will discontinue until the server "
+							       +
+							       "has been restarted. Please resolve the merge conflicts quickly."
+							       + "  \nThank you :)";
+
+							Utils.bash("bash", "-c", "git status");
+							Utils.bash("bash", "-c", "git restore --staged .");
+							Utils.bash("bash", "-c", "git checkout .");
+
+							Utils.sendNotificationToGitHubThread(msg);
+						} catch (Exception x) {
+							Utils.fatalError("unable to send failure notification!", x);
+						}
+					}
+				while (true);
+			}
+		}.start();
+		System.out.println("[" + new Date() + "] Ready.");
+		while (true) {
+			Socket s = serverSocket.accept();
+			new Thread() {
+				public void run() {
```

Comment:
If you implement the `run` methods in distinct classes the `main` function becomes much more readable
```
class SyncThread implements Runnable {
... the github sync run() method ...
}
class ConnectionThread implements Runnable {
... this run() method ...
}
...
new Thread(new SyncThread()).start();
new Thread(new ConnectionThread()).start();
```
Then it is also easier to pull out some of the run-code into methods.